gateway-messages/src/mgs_to_sp.rs

```diff
@@ -48,8 +48,8 @@ pub enum MgsRequest {
     },
     GetPowerState,
     SetPowerState(PowerState),
-    ResetPrepare,
-    ResetTrigger,
+    ResetPrepare(SpComponent),
+    ResetTrigger(SpComponent),
```

Comment:
CI is failing because this makes a backwards-incompatible change to an existing message, which we shouldn't do - in this case, pushing this update would prevent us from resetting existing SPs that don't have this change.

I think the easiest thing to do here is to just add `ResetComponent{Prepare,Trigger}(SpComponent)` variants to the end of `MgsRequest`, and probably add a corresponding response ack message to the end of `SpResponse`.

A question while we're here - for the SP reset is split into prepare/trigger so MGS can get an ack to the "prepare" message. Is that necessary for resetting components?

Reply:
I assumed that the prepare was to be able to quiesce things or reject, e.g. wrong power state. Is the intent of prepare to just ping the SP?

I'm ok with not making a breaking change. A new message and then later deprecation of the old if appropriate is reasonable.

Reply:
> I assumed that the prepare was to be able to quiesce things or reject, e.g. wrong power state. Is the intent of prepare to just ping the SP?

At the moment yes - `control-plane-agent` doesn't really have any means to quiesce things. It could check the power state but doesn't do that either. If we might want to do either of those for component resets (which seems likely), :+1: on keeping the prepare/trigger pair for the new variants.

Reply:
NM. I see the details of prepare.